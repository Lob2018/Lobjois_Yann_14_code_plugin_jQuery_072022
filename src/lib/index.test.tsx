import React from "react";
import renderer from "react-test-renderer";
import Example from '../examples/App'

describe("Dropdown", () => {
  it("renders properly", async () => {
    const component = renderer.create(
      <Example></Example>,
    );  
    let tree = component.toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
  });
});
