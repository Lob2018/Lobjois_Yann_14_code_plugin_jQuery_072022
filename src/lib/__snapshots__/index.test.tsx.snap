// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders properly 1`] = `
<div
  style={
    Object {
      "margin": "15px auto",
      "width": "100%",
    }
  }
>
  <h1>
    React Dropdown Component Library
  </h1>
  <h2>
    1er dropdown : 
  </h2>
  <h2>
    2ème dropdown : 
  </h2>
  <h2>
    3ème dropdown : 
  </h2>
  <label
    id="combo-label1"
  >
    Dropdown with values
  </label>
  <div
    className="combo js-select"
  >
    <div
      aria-activedescendant=""
      aria-controls="listbox-combo-label1"
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-labelledby="combo-label1"
      className="combo-input"
      id="combo-combo-label1"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
      title="Marketing"
    >
      Marketing
    </div>
    <div
      aria-labelledby="combo-label1"
      className="combo-menu"
      id="listbox-combo-label1"
      role="listbox"
      tabIndex={-1}
    >
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label10"
        onClick={[Function]}
        role="option"
        title="Sales"
      >
        Sales
      </div>
      <div
        aria-selected={true}
        className="combo-option option-current"
        id="combo-label11"
        onClick={[Function]}
        role="option"
        title="Marketing"
      >
        Marketing
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label12"
        onClick={[Function]}
        role="option"
        title="Engineering"
      >
        Engineering
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label13"
        onClick={[Function]}
        role="option"
        title="Human Resources"
      >
        Human Resources
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label14"
        onClick={[Function]}
        role="option"
        title="Legal"
      >
        Legal
      </div>
    </div>
  </div>
  <label
    id="combo-label2"
  >
    Dropdown with values and overrided titles
  </label>
  <div
    className="combo js-select"
  >
    <div
      aria-activedescendant=""
      aria-controls="listbox-combo-label2"
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-labelledby="combo-label2"
      className="combo-input"
      id="combo-combo-label2"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
      title="CA"
    >
      California
    </div>
    <div
      aria-labelledby="combo-label2"
      className="combo-menu"
      id="listbox-combo-label2"
      role="listbox"
      tabIndex={-1}
    >
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label20"
        onClick={[Function]}
        role="option"
        title="AL"
      >
        Alabama
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label21"
        onClick={[Function]}
        role="option"
        title="AK"
      >
        Alaska
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label22"
        onClick={[Function]}
        role="option"
        title="AS"
      >
        American Samoa
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label23"
        onClick={[Function]}
        role="option"
        title="AZ"
      >
        Arizona
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label24"
        onClick={[Function]}
        role="option"
        title="AR"
      >
        Arkansas
      </div>
      <div
        aria-selected={true}
        className="combo-option option-current"
        id="combo-label25"
        onClick={[Function]}
        role="option"
        title="CA"
      >
        California
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label26"
        onClick={[Function]}
        role="option"
        title="CO"
      >
        Colorado
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label27"
        onClick={[Function]}
        role="option"
        title="CT"
      >
        Connecticut
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label28"
        onClick={[Function]}
        role="option"
        title="DE"
      >
        Delaware
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label29"
        onClick={[Function]}
        role="option"
        title="DC"
      >
        District Of Columbia
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label210"
        onClick={[Function]}
        role="option"
        title="FM"
      >
        Federated States Of Micronesia
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label211"
        onClick={[Function]}
        role="option"
        title="FL"
      >
        Florida
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label212"
        onClick={[Function]}
        role="option"
        title="GA"
      >
        Georgia
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label213"
        onClick={[Function]}
        role="option"
        title="GU"
      >
        Guam
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label214"
        onClick={[Function]}
        role="option"
        title="HI"
      >
        Hawaii
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label215"
        onClick={[Function]}
        role="option"
        title="ID"
      >
        Idaho
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label216"
        onClick={[Function]}
        role="option"
        title="IL"
      >
        Illinois
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label217"
        onClick={[Function]}
        role="option"
        title="IN"
      >
        Indiana
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label218"
        onClick={[Function]}
        role="option"
        title="IA"
      >
        Iowa
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label219"
        onClick={[Function]}
        role="option"
        title="KS"
      >
        Kansas
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label220"
        onClick={[Function]}
        role="option"
        title="KY"
      >
        Kentucky
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label221"
        onClick={[Function]}
        role="option"
        title="LA"
      >
        Louisiana
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label222"
        onClick={[Function]}
        role="option"
        title="ME"
      >
        Maine
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label223"
        onClick={[Function]}
        role="option"
        title="MH"
      >
        Marshall Islands
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label224"
        onClick={[Function]}
        role="option"
        title="MD"
      >
        Maryland
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label225"
        onClick={[Function]}
        role="option"
        title="MA"
      >
        Massachusetts
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label226"
        onClick={[Function]}
        role="option"
        title="MI"
      >
        Michigan
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label227"
        onClick={[Function]}
        role="option"
        title="MN"
      >
        Minnesota
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label228"
        onClick={[Function]}
        role="option"
        title="MS"
      >
        Mississippi
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label229"
        onClick={[Function]}
        role="option"
        title="MO"
      >
        Missouri
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label230"
        onClick={[Function]}
        role="option"
        title="MT"
      >
        Montana
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label231"
        onClick={[Function]}
        role="option"
        title="NE"
      >
        Nebraska
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label232"
        onClick={[Function]}
        role="option"
        title="NV"
      >
        Nevada
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label233"
        onClick={[Function]}
        role="option"
        title="NH"
      >
        New Hampshire
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label234"
        onClick={[Function]}
        role="option"
        title="NJ"
      >
        New Jersey
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label235"
        onClick={[Function]}
        role="option"
        title="NM"
      >
        New Mexico
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label236"
        onClick={[Function]}
        role="option"
        title="NY"
      >
        New York
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label237"
        onClick={[Function]}
        role="option"
        title="NC"
      >
        North Carolina
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label238"
        onClick={[Function]}
        role="option"
        title="ND"
      >
        North Dakota
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label239"
        onClick={[Function]}
        role="option"
        title="MP"
      >
        Northern Mariana Islands
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label240"
        onClick={[Function]}
        role="option"
        title="OH"
      >
        Ohio
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label241"
        onClick={[Function]}
        role="option"
        title="OK"
      >
        Oklahoma
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label242"
        onClick={[Function]}
        role="option"
        title="OR"
      >
        Oregon
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label243"
        onClick={[Function]}
        role="option"
        title="PW"
      >
        Palau
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label244"
        onClick={[Function]}
        role="option"
        title="PA"
      >
        Pennsylvania
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label245"
        onClick={[Function]}
        role="option"
        title="PR"
      >
        Puerto Rico
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label246"
        onClick={[Function]}
        role="option"
        title="RI"
      >
        Rhode Island
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label247"
        onClick={[Function]}
        role="option"
        title="SC"
      >
        South Carolina
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label248"
        onClick={[Function]}
        role="option"
        title="SD"
      >
        South Dakota
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label249"
        onClick={[Function]}
        role="option"
        title="TN"
      >
        Tennessee
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label250"
        onClick={[Function]}
        role="option"
        title="TX"
      >
        Texas
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label251"
        onClick={[Function]}
        role="option"
        title="UT"
      >
        Utah
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label252"
        onClick={[Function]}
        role="option"
        title="VT"
      >
        Vermont
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label253"
        onClick={[Function]}
        role="option"
        title="VI"
      >
        Virgin Islands
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label254"
        onClick={[Function]}
        role="option"
        title="VA"
      >
        Virginia
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label255"
        onClick={[Function]}
        role="option"
        title="WA"
      >
        Washington
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label256"
        onClick={[Function]}
        role="option"
        title="WV"
      >
        West Virginia
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label257"
        onClick={[Function]}
        role="option"
        title="WI"
      >
        Wisconsin
      </div>
      <div
        aria-selected={false}
        className="combo-option"
        id="combo-label258"
        onClick={[Function]}
        role="option"
        title="WY"
      >
        Wyoming
      </div>
    </div>
  </div>
  <p>
    No data found
  </p>
</div>
`;
